mobile-widgets/qml/DownloadFromDiveComputer.qml

```diff
@@ -348,13 +348,8 @@ Kirigami.Page {
 
 		onVisibleChanged: {
 			comboVendor.currentIndex = comboProduct.currentIndex = comboConnection.currentIndex = -1
-			if (visible) {
-				comboVendor.currentIndex = manager.getDetectedVendorIndex()
-				comboProduct.currentIndex = manager.getDetectedProductIndex(comboVendor.currentText)
-				comboConnection.currentIndex = manager.getMatchingAddress(comboVendor.currentText, comboProduct.currentText)
-				manager.DC_vendor = comboVendor.currentText
-				manager.DC_product = comboProduct.currentText
-				manager.DC_devName = comboConnection.currentText
+			if (visible && PrefDiveComputer.vendor !== "" ) {
+				rememberedDCsGrid.setDC(PrefDiveComputer.vendor1, PrefDiveComputer.product1, PrefDiveComputer.device1)
```

Comment:
but if we don't have a DC remembered, we should still try to guess one... so I'm not sure I understand why you removed that code...

Reply:
Partly to remove stale data.
Example scenario:
Connect DC vi a USB, download data and remove cable.
We now have a saved DC and switching to DC download will give us the correct DC selected.
If we then forget the DCs and go to DC download again we are presented with the vendor of the last used DC. 
This is wrong, if we don't have a DC paired or connected via USB we should be presented with an empty dialog.

Reply:
Seems we will need that code for BT/BLE DCs...
I'll remove this commit as the rest is ready to be merged together with #1707 .

Reply:
Now that I removed it I see what's missing..
Getting this to work will take some time, let's merge these as they are right now and I'll fix the automagic part separately.

Reply:
ok